src/pages/iou/request/step/IOURequestStepScan/types.ts

```diff
@@ -0,0 +1,18 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+
+type IOURequestStepRoute = StackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.SCAN_TAB>;
+
+type IOURequestStepReport = {
+    report: OnyxTypes.Report;
+};
+
+type IOURequestStepTransaction = {
+    transaction: OnyxTypes.Transaction;
+};
```

Comment:
1. Its not needed to seperate those types
2. `StackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.SCAN_TAB>;` is wrong type here you should declare it as `{route: RouteProp<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.SCAN_TAB>;}` 
3. another problem is that HOCs which are used in the component arent migrated to TS yet 
4. We should remove compose usage from a component 
So please do changes which I provided, and add `@ts-expect-error` with explenantion